src/adapter/src/client.rs

```diff
@@ -148,6 +150,22 @@ impl Client {
         Session::new(self.build_info, config, self.metrics().session_metrics())
     }
 
+    /// Preforms an authentication check for the given user.
+    pub async fn authenticate(
+        &self,
+        user: &String,
+        password: &String,
```

Comment:
Given this is the plaintext password and it's plumbed through a number of layers, I feel pretty strongly that we should wrap it in a newtype that purposefully doesn't implement `Debug` or `Display`, we can also use the [`assert_not_impl_all!`](https://docs.rs/static_assertions/1.1.0/static_assertions/macro.assert_not_impl_all.html) macro!

FWIW I realize now that we don't do this for Frontegg App Passwords, but definitely should

Reply:
I agree and would add that this should not be merged without that.